packages/webdriverio/src/utils/interception/devtools.ts

```diff
@@ -36,12 +36,16 @@ export default class DevtoolsInterception extends Interception {
             // responseHeaders and responseStatusCode are only present in Response stage
             // https://chromedevtools.github.io/devtools-protocol/tot/Fetch/#event-requestPaused
             const isRequest = !event.responseHeaders
-            const eventResponseHeaders = event.responseHeaders || []
-            const responseHeaders = eventResponseHeaders.reduce((headers, { name, value }) => {
+
+            event.responseStatusCode ||= 200
+            event.responseHeaders ||= []
+
+            const responseHeaders = event.responseHeaders.reduce((headers, { name, value }) => {
                 headers[name] = value
                 return headers
             }, {} as Record<string, string>)
-            const { requestId, request, responseStatusCode = 200 } = event
```

Comment:
moved this line down, so we could properly modify `event` fields in event callback